pkg/server/commands/listusers/list_users_rpc.go

```diff
@@ -592,14 +642,116 @@ func (l *listUsersQuery) expandExclusion(
 	}
 
 	wildcardKey := tuple.TypedPublicWildcard(req.GetUserFilters()[0].GetType())
-	_, subtractWildcardExists := subtractFoundUsersMap[wildcardKey]
-	for key := range baseFoundUsersMap {
-		if _, isSubtracted := subtractFoundUsersMap[key]; !isSubtracted && !subtractWildcardExists {
-			// Iterate over base users because at minimum they need to pass
-			// but then they are further compared to the subtracted users map.
-			// If users exist in both maps, they are excluded. Only users that exist
-			// solely in the base map will be returned.
-			trySendResult(ctx, tuple.StringToUserProto(key), foundUsersChan)
+
+	_, baseWildcardExists := baseFoundUsersMap[wildcardKey]
+	subtractWildcardUser, subtractWildcardExists := subtractFoundUsersMap[wildcardKey]
+
+	if baseWildcardExists {
```

Comment:
Heads up, the linter is complaining about this set of conditionals.

Reply:
I've inverted it now. The control flow now works under the outer range over the `baseFoundUsersMap`. Originally I just wanted to get the logic of the `expandExclusion` correct, but now taking a second look at it the control flow was a bit repetitive. It's more succint now.